bundles/org.corpus_tools.hexatomic.graph/src/main/java/org/corpus_tools/hexatomic/graph/internal/SaltGraphLayout.java

```diff
@@ -0,0 +1,425 @@
+/*-
+ * #%L
+ * org.corpus_tools.hexatomic.graph
+ * %%
+ * Copyright (C) 2018 - 2019 Stephan Druskat, Thomas Krause
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+package org.corpus_tools.hexatomic.graph.internal;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.ComparisonChain;
+import com.google.common.collect.HashBiMap;
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.Range;
+import java.util.BitSet;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import org.corpus_tools.salt.SALT_TYPE;
+import org.corpus_tools.salt.common.SDocumentGraph;
+import org.corpus_tools.salt.common.STextualDS;
+import org.corpus_tools.salt.common.SToken;
+import org.corpus_tools.salt.core.SNode;
+import org.corpus_tools.salt.core.SRelation;
+import org.corpus_tools.salt.graph.IdentifiableElement;
+import org.corpus_tools.salt.util.DataSourceSequence;
+import org.eclipse.zest.layouts.algorithms.AbstractLayoutAlgorithm;
+import org.eclipse.zest.layouts.dataStructures.InternalNode;
+import org.eclipse.zest.layouts.dataStructures.InternalRelationship;
+
+/**
+ * Hierarchical custom layout for Salt graphs.
+ * This will place dominance nodes in a hierarchy and all tokens as ordered leaves.
+ * Dominance nodes are put on different ranks if they overlap the same token range.
+ * 
+ * @author Thomas Krause
+ *
+ */
+public class SaltGraphLayout extends AbstractLayoutAlgorithm {
+
+  private double averageTokenNodeWidth;
+  private double maxNodeHeight;
+
+  private final double percentMargin = 1.8;
```

Comment:
```suggestion
  private final double percentInterNodeMargin = 1.8;
```

Trying to make it clearer that this is the margin between nodes.

Reply:
This is also something that we can expose later (much later) in the UI to let users control the layout algorithm... (#133)